File: HackerRank/Algorithms/1. Warmup/Time_conversion.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {
    /* Enter your code here.
       Read input from STDIN.
       Print output to STDOUT */
    string str;
    cin >> str;
    if (str[8] == 'A') {
        if (str.substr(0, 2) == "12") {
            str = "00" + str.substr(2, 6);
        }
        else {
            str = str.substr(0, 8);
        }
    }
    else {
        if (str.substr(0, 2) == "12") {
            str = str.substr(0, 8);
        }
        else {
            str = to_string(stoi(str.substr(0, 2)) + 12) + str.substr(2, 6);
        }
    }
    cout << str << endl;
    return 0;
}

File: HackerRank/Algorithms/1. Warmup/Time_conversion.java
import java.util.Scanner;

public class Solution {

    public static void main(String[] args) {
        /* Enter your code here.
           Read input from STDIN.
           Print output to STDOUT.
           Your class should be named Solution. */
        Scanner scan = new Scanner(System.in);
        String time = scan.nextLine();
        scan.close();
        if (time.charAt(8) == 'A') {
            if (time.substring(0, 2).equals("12")) {
                time = "00" + time.substring(2, 8);
            }
            else {
                time = time.substring(0, 8);
            }
        }
        else {
            if (time.substring(0, 2).equals("12")) {
                time = time.substring(0, 8);
            }
            else {
                time = Integer.parseInt(time.substring(0, 2)) + 12 + time.substring(2, 8);
            }
        }
        System.out.println(time);
    }
}
